profile/src/main/java/rdx/works/profile/data/model/extensions/EntityExtensions.kt

```diff
@@ -4,28 +4,89 @@ import rdx.works.core.mapWhen
+            is FactorInstance.Badge.VirtualSource.HierarchicalDeterministic -> {
+                virtualBadge.publicKey.curve == Slip10Curve.SECP_256K1
+            }
 
-        null -> false
+            null -> false
+        }
+    }
+
+val Entity.factorSourceId
+    get() = (this.securityState as SecurityState.Unsecured).unsecuredEntityControl.transactionSigning.factorSourceId
+
+val Entity.derivationPathScheme: DerivationPathScheme
+    get() {
+        val transactionSigning = (this.securityState as SecurityState.Unsecured).unsecuredEntityControl.transactionSigning
+        return when (transactionSigning.badge) {
+            is FactorInstance.Badge.VirtualSource.HierarchicalDeterministic -> {
+                transactionSigning.badge.derivationPath.scheme
+            }
+        }
     }
-}
 
-fun Entity.factorSourceId(): FactorSource.FactorSourceID {
-    return (this.securityState as SecurityState.Unsecured).unsecuredEntityControl.transactionSigning.factorSourceId
+fun Network.Account.updateDerivationPathScheme(derivationPathScheme: DerivationPathScheme): Network.Account {
+    val transactionSigning = (this.securityState as SecurityState.Unsecured).unsecuredEntityControl.transactionSigning
+    return when (transactionSigning.badge) {
+        is FactorInstance.Badge.VirtualSource.HierarchicalDeterministic -> {
+            val updatedBadge =
+                transactionSigning.badge.copy(derivationPath = transactionSigning.badge.derivationPath.copy(scheme = derivationPathScheme))
+            val updatedTransactionSigning = transactionSigning.copy(badge = updatedBadge)
+            val updatedUnsecuredEntityControl =
+                this.securityState.unsecuredEntityControl.copy(transactionSigning = updatedTransactionSigning)
+            val updatedSecurityState = SecurityState.Unsecured(unsecuredEntityControl = updatedUnsecuredEntityControl)
+            copy(securityState = updatedSecurityState)
+        }
+    }
 }
 
+val Entity.hasWrongDerivationPathScheme: Boolean
+    get() {
+        val transactionSigning = (this.securityState as SecurityState.Unsecured).unsecuredEntityControl.transactionSigning
+        return when (transactionSigning.badge) {
+            is FactorInstance.Badge.VirtualSource.HierarchicalDeterministic -> {
+                transactionSigning.badge.derivationPath.isBip44LikePath() &&
+                    transactionSigning.badge.derivationPath.scheme == DerivationPathScheme.CAP_26
+            }
+        }
+    }
+
+val Entity.derivationPathEntityIndex: Int
+    get() {
+        val transactionSigning = (this.securityState as SecurityState.Unsecured).unsecuredEntityControl.transactionSigning
+        return when (transactionSigning.badge) {
+            is FactorInstance.Badge.VirtualSource.HierarchicalDeterministic -> {
+                transactionSigning.badge.derivationPath.derivationPathEntityIndex()
+            }
+        }
+    }
+
```

Comment:
aren't all these extension functions only for `Account`?
If yes then why not to extend them from `Account` and also move them to the `AccountExtensions.kt`?

Reply:
no, not all of them, but some

Reply:
where else are they used then besides the accounts?